platform/packages/platform/src/reply.rs

```diff
@@ -3,59 +3,83 @@
 //! Here are defined wrappers for deserializing such structures.
 
 use prost::Message;
-use serde::{de::DeserializeOwned, Deserialize};
+use serde::de::DeserializeOwned;
 
-use sdk::cosmwasm_std::{from_json, Addr, Api, Binary, Reply, StdError, StdResult};
+use sdk::{
+    cosmos_sdk_proto::cosmwasm::wasm::v1::{
+        MsgExecuteContractResponse, MsgInstantiateContract2Response, MsgInstantiateContractResponse,
+    },
+    cosmwasm_std::{from_json, Addr, Api, Binary, Reply, StdError, StdResult},
+};
 
 pub struct InstantiateResponse<T> {
     pub address: Addr,
-    pub data: Option<T>,
+    pub data: T,
 }
 
-pub fn from_instantiate<T>(api: &dyn Api, reply: Reply) -> StdResult<InstantiateResponse<T>>
-where
-    T: DeserializeOwned,
-{
-    #[derive(Message)]
-    struct ReplyData {
-        #[prost(bytes, tag = "1")]
-        pub address: Vec<u8>,
-        #[prost(bytes, tag = "2")]
-        pub data: Vec<u8>,
+impl InstantiateResponse<Vec<u8>> {
+    fn into_addr(self) -> Addr {
+        self.address
     }
+}
 
-    let response = decode::<ReplyData>(reply)?;
+pub fn from_instantiate_addr_only(api: &dyn Api, reply: Reply) -> StdResult<Addr> {
```

Comment:
[all public functions] we shall return `platform::error::Error` not the cosmwasm one

Reply:
Spawning task is a nice instrument for tracking issues in the code that we have noticed. Here, as in the other cases before, we *introduce* issues.